Modules/vc-module-export/VirtoCommerce.ExportModule.Core/Model/ExportedTypeMetadata.cs

```diff
@@ -8,45 +8,74 @@ namespace VirtoCommerce.ExportModule.Core.Model
 {
     public class ExportedTypeMetadata : ValueObject
     {
+        /// <summary>
+        /// Extended type info. Stores if specific property is entity (true) or not (false)
+        /// </summary>
+        private class ExportTypePropertyInfoEx
+        {
+            public ExportTypePropertyInfo PropertyInfo { get; set; }
+            public bool IsEntity { get; set; }
+        }
+
         public string Version { get; set; }
         public ExportTypePropertyInfo[] PropertiesInfo { get; set; }
 
+        /// <summary>
+        /// Returns basic info about potentially exported fields of entity
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
         public static ExportedTypeMetadata GetFromType<T>()
         {
             var result = new ExportedTypeMetadata();
             var t = typeof(T);
             var passedNodes = new List<MemberInfo>();
-            result.PropertiesInfo = result.GetFromType(t, string.Empty, passedNodes);
+            result.PropertiesInfo = result.GetFromType(t, string.Empty, passedNodes).Where(x => !x.IsEntity).Select(x => x.PropertyInfo).ToArray();
             return result;
         }
 
-        private ExportTypePropertyInfo[] GetFromType(Type t, string baseMemberName, List<MemberInfo> passedNodes)
+        private ExportTypePropertyInfoEx[] GetFromType(Type t, string baseMemberName, List<MemberInfo> passedNodes)
         {
-            var result = new List<ExportTypePropertyInfo>();
+            var result = new List<ExportTypePropertyInfoEx>();
+            var membersForNodes = new List<(MemberInfo MemberInfo, Type NestedType)>();
+
             foreach (var pi in t.GetProperties().Where(x => x.CanRead))
             {
                 if (!passedNodes.Contains(pi))
                 {
-                    var derivedMemberName = $"{baseMemberName}{(baseMemberName.IsNullOrEmpty() ? string.Empty : ".")}{pi.Name}";
                     var nestedType = GetNestedType(pi.PropertyType);
+                    bool isEntity = false;
                     if (nestedType.IsSubclassOf(typeof(Entity)))
                     {
-                        passedNodes.Add(pi);
-                        result.AddRange(GetFromType(nestedType, derivedMemberName, passedNodes));
+                        isEntity = true;
```

Comment:
Please make comments in new line before commented string. That there are less chances to scroll to the right.